_studio/mfx_lib/decode/mjpeg/src/mfx_mjpeg_dec_decode.cpp

```diff
@@ -1534,6 +1534,40 @@ mfxStatus VideoDECODEMJPEGBase_HW::Init(mfxVideoParam *decPar, mfxFrameAllocRequ
     m_pMJPEGVideoDecoder->SetFourCC(request_internal->Info.FourCC);
     m_numPic = 0;
 
+    MFX_CHECK_WITH_ASSERT(!m_pCc, MFX_ERR_UNDEFINED_BEHAVIOR);
+    if (m_needVpp)
+    {
+        bool isD3DToSys = false;
+
+        if(m_vPar.IOPattern & MFX_IOPATTERN_OUT_SYSTEM_MEMORY)
+        {
+            isD3DToSys = true;
+        }
+        else if (m_vPar.IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY)
+        {
+            mfxExtOpaqueSurfaceAlloc *pOpaqAlloc = (mfxExtOpaqueSurfaceAlloc *)GetExtendedBuffer(m_vPar.ExtParam, m_vPar.NumExtParam, MFX_EXTBUFF_OPAQUE_SURFACE_ALLOCATION);
+            if (!pOpaqAlloc)
+                return MFX_ERR_INVALID_VIDEO_PARAM;
+
+            isD3DToSys = !((pOpaqAlloc->Out.Type & MFX_MEMTYPE_SYSTEM_MEMORY) == 0);
+        }
+
+        mfxVideoParam localParams = m_vPar;
+
+        if (m_vPar.mfx.Rotation == MFX_ROTATION_90 || m_vPar.mfx.Rotation == MFX_ROTATION_270)
+        {
+            // Frame allocation is possible inside VideoVppJpegD3D9::Init().
+            // Those frames must have width/height of target image, so the swapping.
+            std::swap(localParams.mfx.FrameInfo.Width, localParams.mfx.FrameInfo.Height);
+            std::swap(localParams.mfx.FrameInfo.CropW, localParams.mfx.FrameInfo.CropH);
+        }
+
+        m_pCc = new VideoVppJpegD3D9(core, isD3DToSys, m_vPar.IOPattern & MFX_IOPATTERN_OUT_OPAQUE_MEMORY);
```

Comment:
according to line 1537 you need to delete the old value first?

Maybe it is time to make ```m_pCc``` a ```unique_ptr```

Reply:
> according to line 1537 you need to delete the old value first?

On the contrary, I checked that `m_pCc` is null. Raw pointers are evil, will do.

Reply:
right, this macro confuses me from time to time... I think we need to change it's name to something more obvious about expected value